lib/stdlib/src/uri_string.erl

```diff
@@ -375,6 +375,80 @@ recompose(Map) ->
     end.
 
 
+%%-------------------------------------------------------------------------
+%% Resolve URIs
+%%-------------------------------------------------------------------------
+-spec resolve(RefURI, BaseURI) -> TargetURI when
+      RefURI :: uri_string() | uri_map(),
+      BaseURI :: uri_string() | uri_map(),
+      TargetURI :: uri_string()
+                 | error().
+resolve(URIMap, BaseURIMap) ->
+    resolve(URIMap, BaseURIMap, []).
+
+
+-spec resolve(RefURI, BaseURI, Options) -> TargetURI when
+      RefURI :: uri_string() | uri_map(),
+      BaseURI :: uri_string() | uri_map(),
+      Options :: [return_map],
+      TargetURI :: uri_string() | uri_map()
+                 | error().
+resolve(URIMap, BaseURIMap, Options) ->
```

Comment:
```erlang
resolve(URIMap, BaseURIMap, Options) ->
    normalize(resolve1(ensure_map_uri(URIMap), BaseURIMap), Options).
```
We can also implement this in a one liner and call ```resolve1/2``` directly. This would mean that the target URI is always normalized. Is there any practical use for producing non-normalized URIs?
E.g.
```
135>  uri_string:resolve(<<"../../../g">>, <<"http://a/b/c/d;p?q">>).
<<"http://a/b/c/../../../g">>

```

Reply:
It should not produce those URIs because the dot segments must be removed according to the algorithm. That's what I was using `normalize` for but it really should just be `remove_dot_segments` as I mentioned. I'll work on it tomorrow or Thursday.

Reply:
Now using the correct function.